### PR TITLE
Delete the verification type system

### PR DESCRIPTION
**Pull request by SingleAccretion**
Contributes to #32648.

This change deletes the verification "type system" (i. e. the `TI_` types) and simplifies the `typeInfo` structure, which now only has two purposes:
1) Carrying signature handles for reference types (used in type updates).
2) Carrying function pointer information (used by delegate construction code).

Expecting some TP improvements.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
[No diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=272991&view=ms.vss-build-web.run-extensions-tab).

@dotnet/jit-contrib

File: src/coreclr/jit/typelist.h [hunk 1]
```diff
@@ -32,41 +32,41 @@ DEF_TP(tn      ,nm        , jitType,     verType, sz,sze,asze, st,al, regTyp,
 */
 
 // clang-format off
-DEF_TP(UNDEF   ,"<UNDEF>" , TYP_UNDEF,   TI_ERROR, 0,  0,  0,   0, 0, VTR_INT,   availableIntRegs,    VTF_ANY)
```
Comment (BruceForstall) [inline src/coreclr/jit/typelist.h:35]: Get rid of `verType` in the two comments above.

**Comment by BruceForstall** [comment]
> Re: asm diffs: it looks like 10% or more of the method contexts fail to replay in this PR, presumably due to a change in JIT/EE interface traffic (thus, the recorded contexts don't have the answers to the questions the JIT is asking). Expected?

Maybe the call to `getClassAttribs` in `verCheckTailCallConstraint`?

**Review by BruceForstall — COMMENTED** [review]
LGTM. One nit.

Someone who know the importer better than me should also look.

**Review by AndyAyersMS — COMMENTED** [review]
Overall this looks great.

So just to confirm--
* the stack type normaliz(s)ation was already redundant?
* the struct type normalization is subsumed by using layouts?

**Comment by SingleAccretion** [comment]
> the stack type normaliz(s)ation was already redundant?
> the struct type normalization is subsumed by using layouts?

Yes and yes.

**Review by AndyAyersMS — APPROVED** [review]

**Comment by AndyAyersMS** [comment]
Thanks again for all the work you are doing to clean up things in the JIT.

We had a decent SPMI collection yesterday (not completely clean, but better than other recent efforts). I will kick off another one since this change introduces a fair number of MC misses.

**Pull request merged**